src/coreComponents/physicsSolvers/fluidFlow/ThermalCompositionalMultiphaseBaseKernels.hpp

```diff
@@ -494,6 +494,269 @@ struct SolidInternalEnergyUpdateKernel
+                                           rankOffset, numComp, dofKey, subRegion, localSolution,
+                                           pressure, temperature, compDens );
+    return ScalingForSystemSolutionKernel::launch< POLICY >( subRegion.size(), kernel );
+  }
+
+};
+
+/******************************** SolutionCheckKernel ********************************/
+
+/**
+ * @class SolutionCheckKernel
+ * @brief Define the kernel for checking the updated solution
+ */
+class SolutionCheckKernel : public isothermalCompositionalMultiphaseBaseKernels::SolutionCheckKernel
+{
+public:
+
+  using Base = isothermalCompositionalMultiphaseBaseKernels::SolutionCheckKernel;
+  using Base::m_numComp;
+  using Base::m_localSolution;
+  using Base::m_scalingFactor;
+
+  static real64 constexpr minTemperature = 273.15;
+
+  /**
+   * @brief Create a new kernel instance
+   * @param[in] allowCompDensChopping flag to allow the component density chopping
+   * @param[in] scalingFactor the scaling factor
+   * @param[in] rankOffset the rank offset
+   * @param[in] numComp the number of components
+   * @param[in] dofKey the dof key to get dof numbers
+   * @param[in] subRegion the subRegion
+   * @param[in] localSolution the Newton update
+   * @param[in] pressure the pressure vector
+   * @param[in] temperature the temperature vector
+   * @param[in] compDens the component density vector
+   */
+  SolutionCheckKernel( integer const allowCompDensChopping,
+                       real64 const scalingFactor,
+                       globalIndex const rankOffset,
+                       integer const numComp,
+                       string const dofKey,
+                       ElementSubRegionBase const & subRegion,
+                       arrayView1d< real64 const > const localSolution,
+                       arrayView1d< real64 const > const pressure,
+                       arrayView1d< real64 const > const temperature,
+                       arrayView2d< real64 const, compflow::USD_COMP > const compDens )
+    : Base( allowCompDensChopping,
+            scalingFactor,
+            rankOffset,
+            numComp,
+            dofKey,
+            subRegion,
+            localSolution,
+            pressure,
+            compDens ),
+    m_temperature( temperature )
+  {}
+
+  /**
+   * @brief Compute the local value of the solution check
+   * @param[in] ei the element index
+   * @param[inout] stack the stack variables
+   */
+  GEOSX_HOST_DEVICE
+  void computeSolutionCheck( localIndex const ei,
+                             StackVariables & stack ) const
+  {
+    Base::computeSolutionCheck( ei, stack, [&] ()
+    {
+      // compute the change in temperature
+      real64 const newTemp = m_temperature[ei] + m_scalingFactor * m_localSolution[stack.localRow + m_numComp + 1];
+      if( newTemp < minTemperature )
+      {
+        stack.localMinVal = 0;
```

Comment:
Why not `stack.localMinVal = newTemp`?

Reply:
is there a throwing strategy on the `0` value ? as negative temperature is cut-off at 0 as well as pressure is IIRC

Reply:
In fact here, `minVal` is an integer flag initialized to 1, and set to zero if the solution is invalid. It may not be very clear I agree, but I could not find a way to work with `bool` in the `RAJA::ReduceMin`.